sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/schemas/namespaces/NamespacesResource.java

```diff
@@ -0,0 +1,404 @@
+                                { "name": "dc2", "replicas": "3" }
+                            ]
+                        }
+                        """),
+              }))
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "201",
+            description = "Namespace created.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = NamespaceDto.class))
+            }),
+        @APIResponse(
+            responseCode = "409",
+            description = "Conflict.",
+            content =
+                @Content(
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class,
+                            example =
+                                """
+                                {
+                                    "code": 409,
+                                    "description": "Create failed: namespace cycling already exists."
+                                }
+                                """))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @POST
+  public Uni<RestResponse<Object>> createNamespace(
+      @Context UriInfo uriInfo,
+      @NotNull(message = "payload not provided") @Valid CreateNamespaceDto body) {
+    String namespaceName = body.name();
+
+    // check existing
+    return namespaceManager
+        .getNamespace(namespaceName)
+
+        // in case there is a namespace, ensure we report as error
+        .map(
+            namespace -> {
+              int code = Response.Status.CONFLICT.getStatusCode();
+              ApiError error =
+                  new ApiError(
+                      "Create failed: namespace %s already exists.".formatted(namespaceName), code);
+              return RestResponse.ResponseBuilder.create(code).entity(error).build();
+            })
+
+        // otherwise if namespace does not exist, manager would fire the error
+        // with DATASTORE_KEYSPACE_DOES_NOT_EXIST
+        .onFailure(
+            t -> {
+              if (t instanceof ErrorCodeRuntimeException ec) {
+                return Objects.equals(
+                    ec.getErrorCode(), ErrorCode.DATASTORE_KEYSPACE_DOES_NOT_EXIST);
+              }
+              return false;
+            })
+        .recoverWithUni(
+            () ->
+                namespaceManager
+                    .createNamespace(namespaceName, getReplication(body))
+                    .map(
+                        created -> {
+                          // TODO should we also ship back the datacenters?
```

Comment:
Same as with replica count, I think it's perfectly fine to add this non-breaking change in

Reply:
Done.